component/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -0,0 +1,170 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.identity.authenticator.x509Certificate;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.application.authentication.framework.AbstractApplicationAuthenticator;
+import org.wso2.carbon.identity.application.authentication.framework.LocalApplicationAuthenticator;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.StepConfig;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+
+/**
+ * Authenticator of X509Certificate
+ */
+public class X509CertificateAuthenticator extends AbstractApplicationAuthenticator implements
+        LocalApplicationAuthenticator {
+    private static Log log = LogFactory.getLog(X509CertificateAuthenticator.class);
+
+    /**
+     * Initialize the process and call servlet .
+     *
+     * @param httpServletRequest    http request
+     * @param httpServletResponse   http response
+     * @param authenticationContext authentication context
+     * @throws AuthenticationFailedException
+     */
+    @Override
+    protected void initiateAuthenticationRequest(HttpServletRequest httpServletRequest,
+                                                 HttpServletResponse httpServletResponse,
+                                                 AuthenticationContext authenticationContext)
+            throws AuthenticationFailedException {
+        String authEndpoint = null;
+        try {
+            authEndpoint = X509CertificateConstants.AUTH_ENDPOINT;
```

Comment:
authEndpoint is an constant right ? Any reason for having initialization and declaration as two lines of code ?